### PR TITLE
Fix for http client

### PR DESCRIPTION
**Pull request by AbdurNawaz**
## Why are these changes needed?

When you add the parameter `http_client` (which is typically an instance of `httpx.Client` or `httpx.AsyncClient`) to llm_config, the method `copy.deepcopy` throws an error as it won't be able to pickle the client object. This is a fix to skip the copying if an http_client parameter is passed.

### Code to reproduce:

```python
import httpx

http_client = httpx.Client()
                          
mixtral_config_list = [                                                                                                
    {
        "base_url": "OPENAI_API_BASE_URL"
        "api_key": "OPENAI_API_KEY",                                                                                       
        "http_client": http_client,
    }
]

#the below line will throw an error
assistant = AssistantAgent("assistant", llm_config={"config_list": config_list})
```

## Related issue number

#2412 

## Checks

- [x] I've included any doc changes needed for https://microsoft.github.io/autogen/. See https://microsoft.github.io/autogen/docs/Contribute#documentation to build and test documentation locally.
- [x] I've added tests (if relevant) corresponding to the changes introduced in this PR.
- [x] I've made sure all auto checks have passed.

### CODE REVIEW

**Comment by ekzhu** [comment]
Thanks for the PR. Given @Gr3atWh173 's comment: https://github.com/microsoft/autogen/issues/2412#issuecomment-2093140918
I think instead of checking for `http_client` key, we should try and raise the error but append a more meaningful error message saying please implement the value classes with `__deepcopy__` method to support deepcopy.

**Review by ekzhu — CHANGES_REQUESTED** [review]
Thanks for the PR. Given @Gr3atWh173 's comment: https://github.com/microsoft/autogen/issues/2412#issuecomment-2093140918
I think instead of checking for `http_client` key, we should try and raise the error but append a more meaningful error message saying please implement the value classes with `__deepcopy__` method to support deepcopy.

**Comment by AbdurNawaz** [comment]
@microsoft-github-policy-service agree

**Comment by AbdurNawaz** [comment]
@ekzhu agreed - I've added the necessary changes and updated the docs. Please review

File: website/docs/topics/llm_configuration.ipynb
Comment (ekzhu) [inline website/docs/topics/llm_configuration.ipynb:1]: Can we change the new section's title slightly to make it about how to add an http client object to the llm_config for proxy.

Comment (AbdurNawaz) [reply website/docs/topics/llm_configuration.ipynb:1]: Added the change in title & updated code to use a sample proxy.

**Review by ekzhu — APPROVED** [review]

**Pull request merged**